browser/components/CodeEditor.js

```diff
@@ -793,8 +808,30 @@ export default class CodeEditor extends React.Component {
     this.updateHighlight(editor, changeObject)
 
     this.value = editor.getValue()
+
+    const { storageKey, noteKey } = this.props
+    const storage = findStorage(storageKey)
     if (this.props.onChange) {
       this.props.onChange(editor)
+      if (storage)
+        sendWakatimeHeartBeat(
+          storage.path,
+          noteKey,
+          storage.name,
+          true,
+          true,
+          false
+        )
+    } else {
+      if (storage)
+        sendWakatimeHeartBeat(
+          storage.path,
+          noteKey,
+          storage.name,
+          false,
+          false,
+          false
+        )
```

Comment:
This part seem to be a bit repeative. Surely we can improve the quality of this code by doing something similar to this:
```js
if (this.props.onChange) {
    this.props.onChange(editor)
}

const isWrite = !!this.props.onChange;
const hasFileChanges = isWrite;

if (storage) {
  sendWakatimeHeartBeat(
    storage.path,
    noteKey,
    storage.name,
    isWrite,
    hasFileChanges,
    false
  )
}
```

Reply:
DONE, It was very old code XD